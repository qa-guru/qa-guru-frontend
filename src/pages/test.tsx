import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
