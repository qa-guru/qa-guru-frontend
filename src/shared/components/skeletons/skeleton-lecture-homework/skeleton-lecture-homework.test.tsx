import { MemoryRouter } from "react-router-dom";
import { render } from "test/utils-test";

import SkeletonLectureHomework from "./skeleton-lecture-homework";

describe("SkeletonLectureHomework", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SkeletonLectureHomework />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
