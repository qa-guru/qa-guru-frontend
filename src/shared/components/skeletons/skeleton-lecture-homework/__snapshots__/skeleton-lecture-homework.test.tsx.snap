// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SkeletonLectureHomework > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1c90ci9"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse css-ysrgr0-MuiSkeleton-root"
    />
  </div>
</DocumentFragment>
`;
